episodes/model-choices.Rmd

```diff
@@ -0,0 +1,418 @@
+
+}")
+```
+
+
+The model is capable of simulating an Ebola type outbreak, but as the model is deterministic, we are not able to explore stochastic variation in the early stages of the outbreak.
+
+
+#### `model_ebola()`
+
+A stochastic SEIHFR (Susceptible, Exposed, Infectious, Hospitalised, Funeral, Removed) model that was developed specifically for infection with Ebola. The model has stochasticity in the passage times between states, which are modelled as Erlang distributions.
+
+The key parameters affecting the transition between states are:
+
++ $R_0$, the basic reproduction number,
++ $\rho^I$, the mean infectious period,
++ $\rho^E$, the mean preinfectious period,
++ $p_{hosp}$ the probability of being transferred to the hospitalised compartment. 
+
+**Note: the functional relationship between the preinfectious  period ($\rho^E$) and the transition rate between exposed and infectious ($\gamma^E$) is $\rho^E = k^E/\gamma^E$ where $k^E$ is the shape of the Erlang distribution. Similarly for the infectious period $\rho^I = k^I/\gamma^I$. See [here](https://epiverse-trace.github.io/epidemics/articles/ebola_model.html#details-discrete-time-ebola-virus-disease-model) for more detail on the stochastic model formulation. ** 
+
+```{r, echo = FALSE, message = FALSE}
+DiagrammeR::grViz("digraph {
+
+  # graph statement
+  #################
+  graph [layout = dot,
+  rankdir = LR,
+  overlap = true,
+  fontsize = 10]
+
+  # nodes
+  #######
+  node [shape = square,
+       fixedsize = true
+       width = 1.3]
+
+       S
+       E
+       I
+       H
+       F
+       R
+
+  # edges
+  #######
+  S -> E [label = ' infection (&beta;)']
+  E -> I [label = ' onset of \ninfectiousness (&gamma; E)']
+  I -> F [label = ' death (funeral) \n(&gamma; I)']
+  F -> R [label = ' safe burial (one timestep) ']
+  I -> H [label = ' hospitalisation (p hosp)']
+  H -> R [label = ' recovery or safe burial \n (&gamma; I)']
+
+  subgraph {
+    rank = same; I; F;
+  }
+  subgraph {
+    rank = same; H; R;
+  }
+}")
+```
+
+The model has additional parameters describing the transmission risk in hospital and funeral settings: 
+
++ $p_{ETU}$, the proportion of hospitalised cases contributing to the infection of susceptibles (ETU = Ebola virus treatment units),
++ $p_{funeral}$, the proportion of funerals at which the risk of transmission is the same as of infectious individuals in the community.
+
+As this model is stochastic, it is the most appropriate choice to explore how variation in numbers of infected individuals in the early stages of an Ebola outbreak. 
+
+
+:::::::::::::::::::::::::::
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+## Challenge : Ebola outbreak analysis 
+
+
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Running the model
+
+You have been tasked to generate initial trajectories of an Ebola outbreak in Guinea. Using `model_ebola()` and the the information detailed below, complete the following tasks:
```

Comment:
Overdispersion - that's a really interesting question. I would guess the possibility exists as the distribution of infectious periods is Erlang distributed (reprex below) so there's potentially lots of individual variation in the number of secondary cases.

Herd immunity - in the sense that final size proportion < 1.0? Yes, reprex shows about 0.5.

``` r
library(epidemics)
library(ggplot2)

dist_infectious_period = epidemics:::prob_discrete_erlang(2, 2 / 12)
barplot(dist_infectious_period)
```

![](https://i.imgur.com/51vMHfn.png)<!-- -->

``` r

# Prepare population and parameters
demography_vector <- 1400000 # small population 1.4 M
contact_matrix <- matrix(1)

# manual case counts divided by pop size rather than proportions as small sizes
# introduce errors when converting to counts in the model code; extra
# individuals may appear
infectious <- 1
exposed <- 10
initial_conditions <- matrix(
  c(demography_vector - infectious - exposed, exposed, infectious, 0, 0, 0) /
    demography_vector,
  nrow = 1
)
rownames(contact_matrix) <- "full_pop"
pop <- population(
  contact_matrix = contact_matrix,
  demography_vector = demography_vector,
  initial_conditions = initial_conditions
)

compartments <- c(
  "susceptible", "exposed", "infectious", "hospitalised", "funeral", "removed"
)

# prepare integer values for expectations on data length
time_end <- 600L
replicates <- 10L

output = model_ebola(pop, time_end = time_end)

ggplot(output[compartment == "removed", ], aes(time, value)) +
  geom_line(aes(group = replicate), linewidth = 0.1) +
  labs(y = "Epidemic size")
```

![](https://i.imgur.com/Hu409OD.png)<!-- -->

<sup>Created on 2024-05-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
> We should also add a note to align with the {epichains} episodes, which models Ebola using a branching process (in practice, the two outputs are probably quite similar, but presumably the {epidemics} model doesn't include overdispersion, but can handle herd immunity if large epidemic occurs?)

Added as an issue #41 

> Overdispersion - that's a really interesting question. I would guess the possibility exists as the distribution of infectious periods is Erlang distributed (reprex below) so there's potentially lots of individual variation in the number of secondary cases.

Could we use [this distribution of secondary cases](https://epiverse-trace.github.io/tutorials-middle/superspreading-simulate.html#from-a-distribution-of-secondary-cases) as a comparison?